### PR TITLE
Blob and Blobstorm Stricter Requirements

### PR DESCRIPTION
**Pull request by jwhitak**
Highly requested.

## What this does
Blob has been firing a little too much too hard so these are some gentle adjustments to get things done. It hasn't actually, but people still complain about 3 blobs in 30 days.

I've removed two security officer types (secoffs and detectives) from the enemy jobs list, since they are unable to get into the laser gun armory to assist in helping fight the blob. They are fancy assistants when the blob strikes. Sorry, but it's true. Instead, I've added the entire engineering department, as they are vital to maintain power and repairs during a blob. Additionally, I've added the Head of Personnel, who can grant access to the guns/cargo, and the QM, who can use the cargo budget to order literally all the guns at a moment's notice.

I've also put a scaling-on-threat enemy jobs requirement on blob. Now if you don't have at least a certain number of enemies for the blob, the blob will not come. This scales on threat level, starting at 4 enemy jobs, and getting reduced to just 1 at 90+ threat. This way, at least some of the crew will be well-equipped for the fight ahead.

Blobstorms have been complained about as well, so I've updated the requirements to match the above. Also why was Roboticist an enemy of blobstorm specifically?

Blob Enemies: Removed Security Officer, Detective, and Roboticist from the enemies list. 
Blob Enemies: Added CE, Atmos Tech, Engineer, QM, and HoP to the enemies list.
Blobstorm (Midround) Pop Requirements: Changed to match roundstart Blob requirements.
Blobstorm (Overmind Random Event): Remains disabled

## Why it's good
Blob won't fire as often on unprepared crew. Or as often, in general.

## Changelog

:cl:
 * tweak: Blob requires specific enemies to trigger now, and more population for a midround blobstorm.

[gamemode] [balance]

### CODE REVIEW

**Comment by west3436** [comment]
key quote earning the thumbs up:
>Blob won't fire as often

**Comment by jwhitak** [comment]
Cancel the threat nerf. It was recently nerfed a month ago, and the nerf was highly effective. We only had blob three times since the nerf. I'll keep the enemy jobs nerf in though.

**Comment by Eneocho** [comment]
Wasn't there a blob tuesdays or similar thing added a while back in #29754?

**Comment by pedr69** [comment]
can you do it for the blobnode meteors(no overmind) too?

**Comment by jwhitak** [comment]
\>Blob Storm is a random event that spawns a whole friggin antag
\>It costs no threat and requires no threat

WHY

**Comment by tinysaturn** [comment]
add botanist as an enemy job
those high potency novaflowers are worth more than an entire engineering team and a cargo supply line combined

**Comment by jwhitak** [comment]
Bandaid for the Blob_storm event. It needs to be removed or reworked to use dynamic threat, and this means touching some of the admin's commands. Alternatively, if I find time, I can look up how to see the round's current threat and check it.

**Comment by Theophrastus1** [comment]
fuck man i have literally been wanting to play blob for MONTHS and you motherfuckers cant let me have it

**Comment by jwhitak** [comment]
As a note, it's not just a nerf. Enemy jobs count double for pop requirements.

The actual nerf is pomf force merging dynamic into exponential threat gen, so you don't get threats high enough for blob as often.

**Comment by jwhitak** [comment]
ACTUALLY I WAS WRONG.

Blob_storm aka the event was disabled. The inherited ``can_start()`` was return 0, so it had 0% to fire. We're good.

**Comment by Eneocho** [comment]
> add botanist as an enemy job those high potency novaflowers are worth more than an entire engineering team and a cargo supply line combined

Amen brother.

**Comment by jwhitak** [comment]
The novaflowers will wreck a blob, sure, but you require an actual skilled botanist with preptime for that. It'd be inappropriate on round start, and questionable at midround. Whereas:

AI can immediately direct the crew and vox announce, being fully annoying about the whole situation. And mass-order guns, bolt open doors for crew to get to said guns, and generally keep an overview of the situation.
Cyborg combat mode lol lmao
Warden/Head of Security/Captain can open the armory.
Quartermaster can mass-order guns, emitters, and even make an engine in a pinch using the full cargo budget.
Head of Personnel is an upgraded quartermaster who can also print all-access IDs.
Station Engineer/CE/Atmos Techs are there to make sure the emitters and power are running, and that the station is atmos-tight.

All have an immediate use, or a vital ongoing use, against the blob. Botanists are hella handy to have, but they usually won't have the ~~200~~198 potency novaflowers ready in a heartbeat like a QM pulling out the card and slapping the SMART LASERS crate as fast as he can while drooling on his console.

So I'm kinda not feeling that botanist should count as a REQUIRED slot to spawn a blob. Not to deny that they're hella handy for beating the monster in. But, say, two botanists at 100 threat will kind of make for a bad blob fight?

**Comment by Eneocho** [comment]
You'd be surprised at just how fast can an experienced botanist whip up the turbo novas, I can reliably get the giga fuck you novas in about 10 minutes.

**Comment by jwhitak** [comment]
You are a skilled botanist player with tons of rounds, and it takes ten minutes for you to setup on a dedicated NEED THIS NOW line. Enemy jobs I feel are more intertwined with the antag. Them existing simply makes it harder for the antag to function, regardless of skill level.

Example, it takes the Captain/HoS/Warden like 20 seconds to run to the armory and open it. Blob's already gotta fight in under 60 seconds against the validhunters. Cargo's got guns in like 3 minutes, even if they're new. Suddenly, you've got an army of guys with guns. These enemy jobs existing hurts the Blob.

Compare to say, scientists. They can get tons of guns printed if they bumrush diamonds on the roid before the lockdown, less than 10 minutes... if they know what they're doing. Average new scientist can barely finish research that quickly, much less have any materials to print the guns. Botany is similar, you have to know to raise the potency, have to know to make novaflowers, have to get the chems, have to wait. All for a single decently powerful... melee weapon. Cargo's bought more gun crates than crew at this point.

Engineers are included because of the massive damage to the station that the blob inflicts, plus the power requirements for all of the guns and the raw lethality of well-done emitters. Having a blob with no engineers SUCKS.

So, basically, botanist can be STRONG against blob, but it's not inherently an enemy job. Nobody will notice that there's no botanist when the blob strikes. They will definitely notice if there's no Sec, Cargo, or Engineering.

**Comment by tinysaturn** [comment]
If you want to get down to it, Cargo is actually pretty useless after ordering a couple of crates of plastic sheets.
The guns that Cargo can order get mogged by Liberators so hard it's not funny. After you print a bunch of those, your next goal should be to make enough rechargers to not bottleneck the crew's counteroffensive.
Botanist being an enemy job was a bit of a tongue in cheek suggestion but honestly, it always horrifies me when playing as Overmind that 1 crewmember is able to make a melee weapon that lets any schmuck 1hit weak blobs and 2hit strong blobs with zero downtime and zero reliance on any cooperation.

**Comment by Dacendeth** [comment]
>Compare to say, scientists. They can g-

Let me stop you right there, a scientist almost never does anything useful on a blob round. Research getting done on a blob round happens once in a blue moon.

**Comment by Eneocho** [comment]
>You are a skilled botanist player with tons of rounds, and it takes ten minutes for you to setup on a dedicated NEED THIS NOW line.

Most of those 10 minutes are spent waiting for the potency to get high enough. You can get regular novas with no xenobotany in about 1 minute if you do it right (L4Z). On the meantime you make ye splices, which takes about 2m, and raise potency, which can take anywhere from 1m to 5m depending on what you are aiming for. You can take some shortcuts here by directly raising the novas potency instead of doing spliceable potency, which can save you 1 minute overall. Best time for decently strong novas (100 potency) in my opinion is about 4 minutes (of which 2 are spent waiting for potency), but you need to work very efficiently and get a bit lucky with the mutation rolls and get a chemist to give you the mutagen.

>So, basically, botanist can be STRONG against blob, but it's not inherently an enemy job. Nobody will notice that there's no botanist when the blob strikes. They will definitely notice if there's no Sec, Cargo, or Engineering.

Fair enough.
